R/module_nested_tabs.R

```diff
@@ -236,94 +240,95 @@ srv_teal_module.teal_module <- function(id,
-        teal_data_to_filtered_data(data_rv(), datanames = active_datanames())
-      })
-    }
-
-    # manage module filters on the module level
-    # important:
-    #   filter_manager_module_srv needs to be called before filter_panel_srv
-    #   Because available_teal_slices is used in FilteredData$srv_available_slices (via srv_filter_panel)
-    #   and if it is not set, then it won't be available in the srv_filter_panel
-    srv_module_filter_manager(modules$label, module_fd = datasets, slices_global = slices_global)
-    filtered_teal_data <- srv_filter_data(
-      "filter_panel",
-      datasets = datasets,
-      active_datanames = active_datanames,
-      data_rv = data_rv,
-      is_active = is_active
-    )
+    module_out <- reactiveVal()
 
-    is_transformer_failed <- reactiveValues()
-    transformed_teal_data <- srv_transform_data(
-      "data_transform",
-      data = filtered_teal_data,
-      transforms = modules$transformers,
-      modules = modules,
-      is_transformer_failed = is_transformer_failed
-    )
-    any_transformer_failed <- reactive({
-      any(unlist(reactiveValuesToList(is_transformer_failed)))
-    })
-    observeEvent(any_transformer_failed(), {
-      if (isTRUE(any_transformer_failed())) {
-        shinyjs::hide("teal_module_ui")
-        shinyjs::hide("validate_datanames")
-        shinyjs::show("transformer_failure_info")
-      } else {
-        shinyjs::show("teal_module_ui")
-        shinyjs::show("validate_datanames")
-        shinyjs::hide("transformer_failure_info")
+    call_once_when(is_active(), {
+      active_datanames <- reactive({
+        .resolve_module_datanames(data = data_rv(), modules = modules)
+      })
+      if (is.null(datasets)) {
+        datasets <- eventReactive(data_rv(), {
+          req(inherits(data_rv(), "teal_data"))
+          logger::log_debug("srv_teal_module@1 initializing module-specific FilteredData")
+          teal_data_to_filtered_data(data_rv(), datanames = active_datanames())
+        })
       }
-    })
 
-    module_teal_data <- reactive({
-      req(inherits(transformed_teal_data(), "teal_data"))
-      all_teal_data <- transformed_teal_data()
-      module_datanames <- .resolve_module_datanames(data = all_teal_data, modules = modules)
-      .subset_teal_data(all_teal_data, module_datanames)
-    })
-
-    srv_validate_reactive_teal_data(
-      "validate_datanames",
-      data = module_teal_data,
-      modules = modules
-    )
+      # manage module filters on the module level
+      # important:
+      #   filter_manager_module_srv needs to be called before filter_panel_srv
+      #   Because available_teal_slices is used in FilteredData$srv_available_slices (via srv_filter_panel)
+      #   and if it is not set, then it won't be available in the srv_filter_panel
+      srv_module_filter_manager(modules$label, module_fd = datasets, slices_global = slices_global)
```

Comment:
When an app is module-specific then `FilteredData` for given module is not created and thus no filters are available for given module yet. This results in a filter-manager-table displaying "NA" for given module. I like the fact that we postpone everything but I would add an info in `ui/srv_filter_manager_panel` that module has not been activated, click a card to do so (or something similar). Message probably should be displayed after hovering an info icon.
 
<img width="1463" alt="image" src="https://github.com/user-attachments/assets/cb955a01-b094-4c11-98b4-0eb6ecb46767">

Reply:
Solved in a503f908a